contracts/non-fungible-tokens/andromeda-crowdfund/src/testing/tests.rs

```diff
@@ -587,6 +662,398 @@ mod test {
             }
         }
     }
+
+    struct PurchaseTierTestCase {
+        name: String,
+        stage: CampaignStage,
+        expected_res: Result<Response, ContractError>,
+        payee: String,
+        start_time: Option<MillisecondsExpiration>,
+        end_time: MillisecondsExpiration,
+        orders: Vec<SimpleTierOrder>,
+        initial_cap: Uint128,
+        funds: Vec<Coin>,
+        denom: Asset,
+    }
+    #[test]
+    fn test_execute_purchase_tiers_native() {
+        // fixed total cost to 100 for valid purchase
+        let env = mock_env();
+        let buyer = "buyer";
+        let test_cases: Vec<PurchaseTierTestCase> = vec![
+            PurchaseTierTestCase {
+                name: "Standard purchase with valid order using native tokens".to_string(),
+                stage: CampaignStage::ONGOING,
+                expected_res: Ok(Response::new()
```

Comment:
nit: In future you can break out the expected responses in to two fields in a test case. One for pass one for error:

```rust
expected_res: Option<BankMsg>
expected_err: Option<ContractError>
```

Then when running the test you call your method without unwrapping:

```rust
let res = execute(deps.as_mut(), env.clone(), info, msg);
if let Some(err) = test.expected_err {
    assert_eq!(res.unwrap_err(), err, "{}: did not receive expected error", test.name)
    continue;
}

let res = res.unwrap();

// .. Rest of assertions
```

You can also then check the response attributes without having to specify them in every test case to reduce redundancy.

What you've done is also fine, just the above might be less work!